bids/layout/index.py

```diff
@@ -164,10 +165,21 @@ def index_files(self):
         """Index all files in the BIDS dataset. """
         self._index_dir(self.root, self.config)
 
-    def index_metadata(self):
+    def index_metadata(self, **filters):
         """Index metadata for all files in the BIDS dataset. """
+        if filters:
+            # ensure we are returning objects
+            filters['return_type'] = 'object'
+            # until 0.11.0, user can specify extension or extensions
+            ext_key = 'extensions' if 'extensions' in filters else 'extension'
+            if filters.get(ext_key, None):
```

Comment:
One final nitpick, otherwise LGTM if tests pass

```suggestion
            if filters.get(ext_key):
```